pkg/streamingpromql/types/hpoint_ring_buffer.go

```diff
@@ -148,9 +176,16 @@ func (b *HPointRingBuffer) Append(p promql.HPoint) error {
 	}
 
 	nextIndex := (b.firstIndex + b.size) % len(b.points)
-	b.points[nextIndex] = p
 	b.size++
-	return nil
+	return &b.points[nextIndex], nil
+}
+
+func (b *HPointRingBuffer) RemoveLastPoint() {
```

Comment:
Would be good to add some tests and a docstring for this

Reply:
I see you've resolved this comment, but it looks like we're still missing tests for this method.